lib/NormalModuleReplacementPlugin.js

```diff
@@ -8,7 +8,7 @@
 const { join, dirname } = require("./util/fs");
 
 /** @typedef {import("./Compiler")} Compiler */
-/** @typedef {function(TODO): void} ModuleReplacer */
```

Comment:
There are far more properties here. You need to use the `ResolveData` from `NormalModuleFactory`

Reply:
Ah sorry, I was looking for a type and did see `ResolveData`, but it seemed like too much for this parameter.

Fixed in [1b20278](https://github.com/webpack/webpack/pull/16419/commits/1b20278fee49d252a1afd5c1b7177fe6e54b8a42)